File: static/unbundled/js/models/crumb.js
/**
 * @class Crumb
 */
class Crumb {
	constructor(i, ll) {
		this.index = i;
		this.lngLat = ll;
		this.body = ""
	}

	setLngLat(lngLat) {
		// should have an error check here
		this.lngLat = lngLat;
	}

	addBody(b) {
		this.body = b;
	}

	length() {
		return this.body.length;
	}

	hasBody() {
		return this.body != "";
	}

	validate() {
		//if lngLat does not have lng and lat return INVALID
		if (this.body === "")
			return CONST.POST.NO_BODY;
		return CONST.POST.VALID;
	}
}

File: models/crumb.py
from const import CONST

def validate_crumb(crumb):
	# if lngLat does not have lng and lat return INVALID
	# if crumb body is too long return invalid
	if "body" not in crumb or crumb["body"] == "":
		return CONST["POST"]["NO_BODY"]
	return CONST["POST"]["VALID"]
	